Tests/NIOTests/DatagramChannelTests.swift

```diff
@@ -791,4 +791,29 @@ final class DatagramChannelTests: XCTestCase {
         XCTAssertNoThrow(try handler.becameUnwritable.futureResult.wait())
         XCTAssertNoThrow(try handler.becameWritable.futureResult.wait())
     }
+
+    func testSetGetPktInfoOption() {
```

Comment:
Can we also add a test that validates that the packet info metadata actually appears and is correct?